cmd/clusterctl/client/cluster/installer.go

```diff
@@ -90,6 +111,63 @@ func installComponentsAndUpdateInventory(components repository.Components, provi
 	return providerInventory.Create(inventoryObject)
 }
 
+// waitForReady waits till the installed components are ready.
+func (i *providerInstaller) waitForReady(opts InstallOptions) error {
+	// If we dont have to wait for providers to be installed
+	// return early.
+	if !opts.WaitProviders {
+		return nil
+	}
```

Comment:
What about moving this into Install so we can avoid to have `waitForReady` as a wrapper for `waitManagerDeploymentsReady`

Reply:
Based on the comment you had previously about having a `waitForProvidersReady` function I think it would be nice if we rename the `waitForReady` function to `waitForProvidersReady` and have it call `waitManagerDeploymentsReady`. This way the install function reads well and our intentions of what want to do are clear without surfacing how we are doing it which, as you previously mentioned, can change.

This will be address as part of https://github.com/kubernetes-sigs/cluster-api/pull/4934#discussion_r683776625

WDYT?